### PR TITLE
sys-fs/e2fsprogs: install all libraries unconditionally

### PR DESCRIPTION
**Pull request by SoapGentoo**
Closes: https://bugs.gentoo.org/837668
Signed-off-by: David Seifert <soap@gentoo.org>

### CODE REVIEW

**Comment by SoapGentoo** [comment]
diff for USE="-tools":

```
--- before
+++ after
@@ -1,38 +1,74 @@
-/var/tmp/portage/sys-fs/e2fsprogs-1.46.5/image
+/var/tmp/portage/sys-fs/e2fsprogs-1.46.5-r1/image
 └── usr
     ├── bin
     │   ├── compile_et
     │   └── mk_cmds
     ├── include
     │   ├── com_err.h -> et/com_err.h
+    │   ├── e2p
+    │   │   └── e2p.h
     │   ├── et
     │   │   └── com_err.h
-    │   └── ss
-    │       ├── ss_err.h
-    │       └── ss.h
+    │   ├── ext2fs
+    │   │   ├── bitops.h
+    │   │   ├── ext2_err.h
+    │   │   ├── ext2_ext_attr.h
+    │   │   ├── ext2_fs.h
+    │   │   ├── ext2fs.h
+    │   │   ├── ext2_io.h
+    │   │   ├── ext2_types.h
+    │   │   ├── ext3_extents.h
+    │   │   ├── hashmap.h
+    │   │   ├── qcow2.h
+    │   │   └── tdb.h
+    │   ├── i686-pc-linux-gnu
+    │   │   └── ext2fs
+    │   │       └── ext2_types.h
+    │   ├── ss
+    │   │   ├── ss_err.h
+    │   │   └── ss.h
+    │   └── x86_64-pc-linux-gnu
+    │       └── ext2fs
+    │           └── ext2_types.h
     ├── lib
     │   ├── libcom_err.so -> libcom_err.so.2
     │   ├── libcom_err.so.2 -> libcom_err.so.2.1
     │   ├── libcom_err.so.2.1
+    │   ├── libe2p.so -> libe2p.so.2
+    │   ├── libe2p.so.2 -> libe2p.so.2.3
+    │   ├── libe2p.so.2.3
+    │   ├── libext2fs.so -> libext2fs.so.2
+    │   ├── libext2fs.so.2 -> libext2fs.so.2.4
+    │   ├── libext2fs.so.2.4
     │   ├── libss.so -> libss.so.2
     │   ├── libss.so.2 -> libss.so.2.0
     │   ├── libss.so.2.0
     │   └── pkgconfig
     │       ├── com_err.pc
+    │       ├── e2p.pc
+    │       ├── ext2fs.pc
     │       └── ss.pc
     ├── lib64
     │   ├── libcom_err.so -> libcom_err.so.2
     │   ├── libcom_err.so.2 -> libcom_err.so.2.1
     │   ├── libcom_err.so.2.1
+    │   ├── libe2p.so -> libe2p.so.2
+    │   ├── libe2p.so.2 -> libe2p.so.2.3
+    │   ├── libe2p.so.2.3
+    │   ├── libext2fs.so -> libext2fs.so.2
+    │   ├── libext2fs.so.2 -> libext2fs.so.2.4
+    │   ├── libext2fs.so.2.4
     │   ├── libss.so -> libss.so.2
     │   ├── libss.so.2 -> libss.so.2.0
     │   ├── libss.so.2.0
     │   └── pkgconfig
     │       ├── com_err.pc
+    │       ├── e2p.pc
+    │       ├── ext2fs.pc
     │       └── ss.pc
     └── share
         ├── doc
-        │   └── e2fsprogs-1.46.5
+        │   └── e2fsprogs-1.46.5-r1
         │       ├── ChangeLog.bz2
         │       └── README.bz2
         ├── et
@@ -48,4 +84,4 @@
             ├── ct_c.awk
             └── ct_c.sed
 
-17 directories, 31 files
+23 directories, 61 files
```

File: sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild [hunk 1]
```diff
@@ -101,50 +97,42 @@ multilib_src_configure() {
...
-		# /usr/lib/, and install linker scripts to /usr/lib/.
-		gen_usr_ldscript -a e2p ext2fs
+		emake -C lib/ext2fs V=1 DESTDIR="${D}" install
+		emake -C lib/e2p V=1 DESTDIR="${D}" install
 	fi
 
-	gen_usr_ldscript -a com_err ss $(usex kernel_linux '' 'uuid blkid')
+	# Move shared libraries to /lib/, install static libraries to
+	# /usr/lib/, and install linker scripts to /usr/lib/.
+	gen_usr_ldscript -a com_err ss ext2fs e2p $(usex kernel_linux '' 'uuid blkid')
```
Comment (thesamesam) [inline sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild:135]: usex can go actually, we don't support eg Gentoo/fbsd anymore, and prefix won't be concerned with split usr booting issues

File: sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild [hunk 2]
```diff
@@ -101,50 +97,42 @@ multilib_src_configure() {
...
-		emake \
-			STRIP=: \
-			DESTDIR="${D}" \
-			install
-
-		# Move shared libraries to /lib/, install static libraries to
-		# /usr/lib/, and install linker scripts to /usr/lib/.
-		gen_usr_ldscript -a e2p ext2fs
+		emake -C lib/ext2fs V=1 DESTDIR="${D}" install
+		emake -C lib/e2p V=1 DESTDIR="${D}" install
```
Comment (thesamesam) [inline sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild:130]: so none of the libs link against util-Linux?

Comment (SoapGentoo) [reply sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild:130]: NEEDED:
```
/usr/lib64/libext2fs.so.2.4 libcom_err.so.2,libc.so.6
/usr/lib64/libss.so.2.0 libcom_err.so.2,libc.so.6
/usr/lib64/libcom_err.so.2.1 libc.so.6,ld-linux-x86-64.so.2
/usr/lib64/libe2p.so.2.3 libc.so.6
/usr/lib/libext2fs.so.2.4 libcom_err.so.2,libc.so.6
/usr/lib/libss.so.2.0 libcom_err.so.2,libc.so.6
/usr/lib/libcom_err.so.2.1 libc.so.6,ld-linux.so.2
/usr/lib/libe2p.so.2.3 libc.so.6
```

**Review by thesamesam — APPROVED** [review]

File: sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild [hunk 3]
```diff
@@ -12,16 +12,22 @@ SRC_URI="https://www.kernel.org/pub/linux/kernel/people/tytso/e2fsprogs/v${PV}/$
...
 	nls? ( virtual/libintl )
-	tools? ( >=sys-apps/util-linux-2.16 )"
-DEPEND="${RDEPEND}"
-BDEPEND="virtual/pkgconfig
+	tools? ( sys-apps/util-linux )"
+# For testing lib/ext2fs, lib/support/libsupport.a is required, which
+# unconditionally includes '<blkid/blkid.h>' from sys-apps/util-linux.
+DEPEND="
+	${RDEPEND}
+	test? ( sys-apps/util-linux )"
```
Comment (floppym) [inline sys-fs/e2fsprogs/e2fsprogs-1.46.5-r1.ebuild:28]: Needs MULTILIB_USEDEP.

**Review by mgorny — APPROVED** [review]
:+1:

**Pull request closed without merging**